api-put-object.go

```diff
@@ -53,6 +53,8 @@ type PutObjectOptions struct {
 	LegalHold               LegalHoldStatus
 	SendContentMd5          bool
 	DisableMultipart        bool
+	ReplicationVersionID    string
```

Comment:
@poornas this should be documented as to be carefully used by applications which know what they are trying to do.

WE should also articulate that this is an extension to versioning implementation by AWS.